ee/src/main/java/org/jboss/as/ee/component/deployers/ModuleJndiBindingProcessor.java

```diff
@@ -304,6 +313,20 @@ public static boolean equals(Object one, Object two) {
         return one == two || (one != null && one.equals(two));
     }
 
+    private static boolean isConcreteClass(ClassInfo classInfo) {
+        return !Modifier.isAbstract(classInfo.flags()) && !Modifier.isInterface(classInfo.flags());
+    }
+
+    private static boolean hasConcreteSubclass(CompositeIndex index, ClassInfo classInfo) {
+        final Set<ClassInfo> subclasses;
+        if (Modifier.isInterface(classInfo.flags())) {
+            subclasses = index.getAllKnownImplementors(classInfo.name());
+        } else {
+            subclasses = index.getAllKnownSubclasses(classInfo.name());
+        }
+        return subclasses.stream().anyMatch(ModuleJndiBindingProcessor::isConcreteClass);
```

Comment:
Please use a for loop. The class metadata memory cost of a method reference vs a for loop is too high for us to be using this code pattern in server code.